### ISSUE TITLE
Docker image cozy/cozy-app-dev 1.2.8 does not work well when behind a corporate proxy

### ISSUE THREAD

**Issue by klehelley** [cozy/cozy-stack#1925]
I have been following the [How to create your first Cozy application](https://docs.cozy.io/en/tutorials/app/) tutorial but when running `yarn start`, instead of the Docker container starting successfully, it fails with the following messages:

    Cozy stack (docker): checking that :8080 is free... Cozy stack (docker): 
    Cozy stack (docker): error: cozy-stack is already running on :8080
    Cozy stack (docker): kill: usage: kill [-s sigspec | -n signum | -sigspec] pid | jobspec ... or kill -l [sigspec]

I have tracked down the cause of the issue and when starting the services, the `cozy-app-dev.sh` script runs command `curl -s --max-time 1 -XGET ":8080"`. ~It looks like `curl` in this case uses the network interface IP address instead of defaulting to `localhost` or `127.0.0.1`. That IP address seems to be dynamic.~ *Edit : in fact as the FQDN/IP part of the URL is missing, it cannot match anything in `no_proxy`, and thus systematically tries to use the proxy when one is defined.*

Now I have configured Docker to use my company's proxy in `~/.docker/config.json`, with `localhost` and `127.0.0.1` being configured to not use the proxy. But because of the way it is called, `curl` sends a request to the proxy, which does respond. As this is not what `cozy-app-dev.sh` is expecting, the script ends.

It would seem that changing line 79 in that file as follows would do the trick:

    - check_not_running ":${COZY_STACK_PORT}" "cozy-stack"
    + check_not_running "localhost:${COZY_STACK_PORT}" "cozy-stack"

I have taken the time to test that change locally by cloning this repository and applying the change, and while it does fix that particular issue, it fails later (visibly the call to `cozy-stack instances add ...` line 168 of `cozy-app-dev.sh`).

    Cozy stack (docker): checking that localhost:8080 is free... Cozy stack (docker): ok
    waiting for couchdb...Cozy stack (docker): ok
    checking couchdb on http://localhost:5984/... Cozy stack (docker): ok
    Cozy stack (docker): starting cozy-stack with /data/cozy-storage...
    Cozy stack (docker): 
    creating instance cozy.tools:8080 using passphrase "cozy"... Cozy stack (docker):                            !! DEVELOPMENT RELEASE !!
    You are running a development release which may deactivate some very important
    security features. Please do not use this binary as your production server.
    
    Cozy stack (docker): error: 
    Failed to create instance for domain cozy.tools:8080
    Error: Post http://localhost:6060/instances?Apps=&ContextName=dev&DiskQuota=0&Domain=cozy.tools%3A8080&DomainAliases=localhost%3A8080&Email=dev%40cozy.io&Locale=&Passphrase=cozy&PublicName=Jane+Doe&Settings=&SwiftCluster=0&TOSSigned=&Timezone=&UUID=: dial tcp 127.0.0.1:6060: connect: connection refused 1

However I have no idea whether that last error is related to a proxy issue or is caused by something else...

**Comment by nono** [comment]
> However I have no idea whether that last error is related to a proxy issue or is caused by something else...

I think it is related to the proxy. Here, the script is doing a request to http://localhost:6060/ (which the admin port of cozy-stack) to register an instance for the domain cozy.tools:8080, and it looks like the HTTP request is not routed to the stack but to somewhere else that refuses the connection.

**Comment by klehelley** [comment]
By installing `tcpdump` in the container and capturing network traffic when the entrypoint script is launched, I managed to find which query is causing the issue: the call to `http://cozy.tools:8080/version/`. As `cozy.tools` was not defined in `no_proxy`, the request was going through the corporate proxy, which obviously was not responding correctly. By fixing that I managed to have the container correctly set up and started.

So it looks like the following are all it takes to have these development tools working behind a corporate proxy:

1. Have `cozy-app-dev.sh` use `curl` to call `"localhost:${COZY_STACK_PORT}"` instead of `":${COZY_STACK_PORT}"`. Should I submit a PR for that?
2. Make sure `no_proxy` is correctly configured in the container, by including `localhost`, `127.0.0.1` (not sure if this one is actually necessary) and `cozy.tools`. I guess that should be documented somewhere.

For reference, the minimalized `~/.docker/config.json` file corresponding to the proxy:

    {
        "proxies": {
            "default": {
                "httpProxy": "http://CORPORATE_PROXY",
                "httpsProxy": "http://CORPORATE_PROXY",
                "noProxy": "localhost,127.0.0.1,cozy.tools"
            }
        }
    }
